jaxrs/resteasy-jaxrs/src/main/java/org/jboss/resteasy/specimpl/ResteasyUriBuilder.java

```diff
@@ -723,24 +776,10 @@ public URI build(Object... values) throws IllegalArgumentException, UriBuilderEx
 
    protected URI buildFromValues(boolean encodeSlash, boolean encoded, Object... values)
    {
-      List<String> params = getPathParamNamesInDeclarationOrder();

```

Comment:
Retrieving URI template parameters to check if each of them is bound to a value forced us to execute heavy java.util.regex.Matcher stuff two times while they can be done only once in buildString(...) method.
Actually in getPathParamNamesInDeclarationOrder() method "PathHelper.URI_PARAM_PATTERN.matcher(PathHelper.replaceEnclosedCurlyBraces(string)).find()" is called	 for each scheme, userInfo, host,path, query and fragment properties. 
Same calls are made in buildString(...) method for same properties. We can do it once for all in buildString(...) and perform checks to throw exceptions if needed with respect of the specification requirements.
By the way doing that will keep us consistent with all buildFromEncodedMap() and buildFromMap() methods where all checks are already performed by buildString(...) method.